bundle/config/mutator/resolve_resource_references.go

```diff
@@ -0,0 +1,61 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/resolvers"
+	"github.com/databricks/cli/libs/log"
+)
+
+var separator string = ":"
+
+type resolveResourceReferences struct {
+	resolvers map[string](resolvers.ResolverFunc)
+}
+
+func ResolveResourceReferences() bundle.Mutator {
+	return &resolveResourceReferences{
+		resolvers: resolvers.Resolvers(),
+	}
+}
+
+func (m *resolveResourceReferences) Apply(ctx context.Context, b *bundle.Bundle) error {
+	for k, v := range b.Config.Variables {
```

Comment:
I think we can/should parallelize these lookups. You could use `sync.WaitGroup` (also see `libs/sync` for parallelization). They are inherently independent. Can also be a follow up.